FirebaseAI/Sources/GenerateContentResponse.swift

```diff
@@ -299,6 +302,123 @@ public struct PromptFeedback: Sendable {
+  /// [Service Specific Terms](https://cloud.google.com/terms/service-terms) for *Grounding with
+  /// Google Search*.
+  public let searchEntryPoint: SearchEntryPoint?
+
+  /// A struct representing the Google Search entry point.
+  ///
+  /// > Important: If using Grounding with Google Search, you are required to comply with the
+  /// [Service Specific Terms](https://cloud.google.com/terms/service-terms) for *Grounding with
+  /// Google Search*.
+  @available(iOS 15.0, macOS 12.0, macCatalyst 15.0, tvOS 15.0, watchOS 8.0, *)
+  public struct SearchEntryPoint: Sendable {
+    /// An HTML/CSS snippet that can be embedded in your app. The snippet is designed to avoid
+    /// undesired interaction with the rest of the page's CSS.
+    ///
+    /// To ensure proper rendering, it's recommended to display this content within a `WKWebView`.
+    public let renderedContent: String?
+  }
+
+  /// Represents a chunk of retrieved data that supports a claim in the model's response. This is
+  /// part
+  /// of the grounding information provided when grounding is enabled.
+  ///
+  /// > Important: If using Grounding with Google Search, you are required to comply with the
+  /// [Service Specific Terms](https://cloud.google.com/terms/service-terms) for *Grounding with
+  /// Google Search*.
+  @available(iOS 15.0, macOS 12.0, macCatalyst 15.0, tvOS 15.0, watchOS 8.0, *)
+  public struct GroundingChunk: Sendable {
+    /// Contains details if the grounding chunk is from a web source.
+    public let web: WebGroundingChunk?
+  }
+
+  /// A grounding chunk sourced from the web.
+  ///
+  /// > Important: If using Grounding with Google Search, you are required to comply with the
+  /// [Service Specific Terms](https://cloud.google.com/terms/service-terms) for *Grounding with
+  /// Google Search*.
+  @available(iOS 15.0, macOS 12.0, macCatalyst 15.0, tvOS 15.0, watchOS 8.0, *)
+  public struct WebGroundingChunk: Sendable {
+    /// The URI of the retrieved web page.
+    public let uri: String?
+    /// The title of the retrieved web page.
+    public let title: String?
+    /// The domain of the original URI from which the content was retrieved (e.g., `example.com`).
+    ///
+    /// This field is only populated when using the Vertex AI Gemini API.
+    public let domain: String?
+  }
+
+  /// Provides information about how a specific segment of the model's response is supported by the
+  /// retrieved grounding chunks.
+  ///
+  /// > Important: If using Grounding with Google Search, you are required to comply with the
+  /// [Service Specific Terms](https://cloud.google.com/terms/service-terms) for *Grounding with
+  /// Google Search*.
+  @available(iOS 15.0, macOS 12.0, macCatalyst 15.0, tvOS 15.0, watchOS 8.0, *)
+  public struct GroundingSupport: Sendable {
+    /// Specifies the segment of the model's response content that this grounding support pertains
+    /// to.
+    public let segment: Segment?
```

Comment:
> Do all of the new optional fields need to be optional?
-- @paulb777 

Based on the documentation for GroundingSupport above
> Provides information about how a specific **segment** of the model's response is supported by the retrieved grounding chunks.

Does it make sense for a developer to ever get a `GroundingSupport` where `segment` is `nil`?

Reply:
Good point. It does not make sense for a developer to get a `GroundingSupport` where `segment` is nil. 

This should be non-optional, and we should remove the `GroundingSupport` from the `GroundingMetadata.groundingSupports` list if it's segment is missing.

Reply:
Changed in https://github.com/firebase/firebase-ios-sdk/pull/15014/commits/a63057aa3d2f5fd1833f35c70ce448bd457126b4